src/mono/wasm/debugger/BrowserDebugProxy/ExpressionEvaluator.cs

```diff
@@ -272,50 +290,69 @@ private string ConvertJSToCSharpLocalVariableAssignment(string idName, JToken va
-            {
-                JObject value = await resolver.Resolve(var.Identifier.Text, token);
-                if (value == null)
-                    throw new ReturnAsErrorException($"The name {var.Identifier.Text} does not exist in the current context", "ReferenceError");
-
-                values.Add(value);
-            }
-
-            return values;
+            JObject value = await resolver.Resolve(identifier.Identifier.Text, token);
+            if (value == null)
+                throw new ReturnAsErrorException($"The name {identifier.Identifier.Text} does not exist in the current context", "ReferenceError");
+            return value;
         }
 
-        private static async Task<IList<JObject>> ResolveMethodCalls(IEnumerable<InvocationExpressionSyntax> methodCalls, Dictionary<string, JObject> memberAccessValues, MemberReferenceResolver resolver, CancellationToken token)
+        private static async Task<(IList<JObject>, IList<JObject>, IList<JObject>)> ResolveMethodCalls(ExpressionSyntaxReplacer replacer, MemberReferenceResolver resolver, CancellationToken token)
         {
-            var values = new List<JObject>();
-            foreach (InvocationExpressionSyntax methodCall in methodCalls)
+            var methodCallValues = new List<JObject>();
+            // used for replacing method call on primitive:
+            var maesValues = new List<JObject>();
+            var identifierValues = new List<JObject>();
+            foreach (InvocationExpressionSyntax methodCall in replacer.methodCall.ToArray())
             {
-                JObject value = await resolver.Resolve(methodCall, memberAccessValues, token);
+                (JObject rootObject, string methodName) = await resolver.ResolveInvokationInfo(methodCall, token);
+                if (rootObject == null)
+                    throw new ReturnAsErrorException($"Failed to resolve root object for {methodCall}", "ReferenceError");
+                // primitives don't have objectId
+                if (!DotnetObjectId.TryParse(rootObject?["objectId"]?.Value<string>(), out DotnetObjectId objectId) &&
+                     new string[] { "string", "number", "boolean", "symbol" }.Contains(rootObject?["type"]?.Value<string>()))
+                {
+                    await ReplaceMethodCall(methodCall);
+                    continue;
+                }
+                JObject value = await resolver.Resolve(objectId, methodName, methodCall, replacer.memberAccessValues, token);
                 if (value == null)
                     throw new ReturnAsErrorException($"Failed to resolve member access for {methodCall}", "ReferenceError");
+                methodCallValues.Add(value);
+            }
+            return (methodCallValues, maesValues, identifierValues);
 
-                values.Add(value);
+            async Task ReplaceMethodCall(InvocationExpressionSyntax method)
+            {
+                replacer.methodCall.Remove(method);
```

Comment:
Add a comment like:
```csharp
                /*
                    Instead of invoking the method on the primitive type ourselves,
                    we emit a local for the primitive, and emit the the method
                    itself in the script. For example:

                    double test_propUlong_2c64c = 12;
                    return (test_propUlong_2c64c.ToString());
                */
```